File: HW1/HW1.Calculator/Calculator.cs
using System;
using System.ComponentModel;
using System.Reflection;

namespace HW1.Calculator
{
    public class Calculator
    {
        public enum Operation
        {
            [Description("+")]
            Plus,
            [Description("-")]
            Minus,
            [Description("*")]
            Multiply,
            [Description("/")]
            Divide,
            IncorrectOperation
        };

        public static string GetDescription(Enum operation)
        {
            Type type = operation.GetType();

            MemberInfo[] memInfo = type.GetMember(operation.ToString());
            if (memInfo != null && memInfo.Length > 0)
            {
                object[] attrs = memInfo[0].GetCustomAttributes(typeof(DescriptionAttribute), false);
                if (attrs != null && attrs.Length > 0)
                    return ((DescriptionAttribute) attrs[0]).Description;
            }

            return operation.ToString();
        }
        public static double Calculate(int val1, Operation operation, int val2)
        {
            return operation switch
            {
                Operation.Plus => val1 + val2,
                Operation.Minus => val1 - val2,
                Operation.Multiply => val1 * val2,
                Operation.Divide => val1 / val2
            };
        }
    }
}

File: HW6/HW6/Calculator.fs
module HW6.Calculator
open HW6.Expression
open HW6.Operation;
let tryCalculate expression =
        match expression.Operation with
        |Operation.Plus -> Ok (expression.V1 + expression.V2)
        |Operation.Minus -> Ok (expression.V1 - expression.V2)
        |Operation.Multiply-> Ok (expression.V1 * expression.V2)
        |Operation.Divide -> match expression.V2 with
                        | 0m -> Error "Division by zero"
                        | _ -> Ok (expression.V1 / expression.V2)


